### ISSUE TITLE
Composables with explicit import are not keyed

### ISSUE THREAD

**Issue by huang-julien** [nuxt/nuxt#28642]
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v20.11.0
- Nuxt Version:     3.12.4
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.3.0
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-cqupsh?file=composables%2Ftest.ts,app.vue,nuxt.config.ts,package-lock.json

### Describe the bug

Composables that are explicitly imported are not keyed even when the source is not specified

**Comment by danielroe** [comment]
I think you just need to provide `source`: https://stackblitz.com/edit/nuxt-starter-tdftk6.

So, two possible improvements:

1. mark `source` required.
2. handle resolving aliases, so `~/composables/test` works as well.

**Comment by Flo0806** [comment]
@danielroe can also be closed - `source` is working as well.

**Comment by danielroe** [comment]
yes, we now handle alias as part of the work done by @cernymatej IIRC.

we don't require `source` (except in types) as that would be breaking, but will be doing that for nuxt v5

nuxtathon closed @Flo0806
